graphql-server-typescript example does not work

# graphql-server-typescript example does not work

## Description

Page https://github.com/accounts-js/accounts/tree/master/examples/graphql-server-typescript states "This example demonstrate how to use accounts-js."

Following instructions...

```bash
git clone https://github.com/accounts-js/examples.git
cd examples/graphql-server-typescript
yarn install
```

...leads to errors and warnings.

Additional instructions...

>Start the app.
>
>Visit http://localhost:4000/
>
>yarn start

... mislead the user to first open the server address in a browser and then start the server.

## Importance

Examples should provide users with working solution without any problems. Especially examples as basic as this one.

## Related

graphql-server-typeorm-postgres does not work either. I did not try other examples but I doubt they work.

## git clone result


```bash
prishub@ubuntu-one:~/dev/xxx/web/accounts-js$ git clone https://github.com/accounts-js/accounts.git
Cloning into 'accounts'...
remote: Enumerating objects: 32, done.
remote: Counting objects: 100% (32/32), done.
remote: Compressing objects: 100% (20/20), done.
remote: Total 15721 (delta 14), reused 20 (delta 7), pack-reused 15689
Receiving objects: 100% (15721/15721), 7.63 MiB | 2.10 MiB/s, done.
Resolving deltas: 100% (11142/11142), done.
prishub@ubuntu-one:~/dev/xxx/web/accounts-js$ cd accounts/examples/graphql-server-typescript/
```


## yarn install result


```bash
prishub@ubuntu-one:~/dev/xxx/web/accounts-js/accounts/examples/graphql-server-typescript$ yarn install
yarn install v1.17.3
[1/4] Resolving packages...
warning Resolution field "graphql-tools@4.0.5" is incompatible with requested version "graphql-tools@4.0.4"
warning Resolution field "graphql-tools@4.0.5" is incompatible with requested version "graphql-tools@4.0.4"
warning Resolution field "@types/node@12.7.4" is incompatible with requested version "@types/node@^10.1.0"
[2/4] Fetching packages...
info fsevents@2.0.7: The platform "linux" is incompatible with this module.
info "fsevents@2.0.7" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@1.2.9: The platform "linux" is incompatible with this module.
info "fsevents@1.2.9" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning " > graphql-tools@4.0.5" has unmet peer dependency "graphql@^0.13.0 || ^14.0.0".
warning "graphql-tools > apollo-link@1.2.12" has unmet peer dependency "graphql@^0.11.3 || ^0.12.3 || ^0.13.0 || ^14.0.0".
warning "graphql-tools > apollo-utilities@1.3.2" has unmet peer dependency "graphql@^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0".
warning " > ts-jest@24.0.2" has unmet peer dependency "jest@>=24 <25".
warning " > @accounts/express-session@0.19.0" has unmet peer dependency "express@^4.16.3".
warning " > @accounts/express-session@0.19.0" has unmet peer dependency "express-session@^1.15.6".
warning " > @accounts/graphql-api@0.19.0" has unmet peer dependency "@graphql-modules/core@0.7.9".
warning " > @accounts/graphql-api@0.19.0" has unmet peer dependency "graphql-tag@^2.10.0".
warning "workspace-aggregator-58e386bf-df20-49ab-83f3-a64598680545 > @accounts/graphql-api > graphql-toolkit@0.5.12" has unmet peer dependency "graphql-tag-pluck@^0.8.3".
warning "workspace-aggregator-58e386bf-df20-49ab-83f3-a64598680545 > @accounts/graphql-client > graphql-tag@2.10.1" has unmet peer dependency "graphql@^0.9.0 || ^0.10.0 || ^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0".
warning "workspace-aggregator-58e386bf-df20-49ab-83f3-a64598680545 > @examples/react-graphql-typescript > react-apollo@2.5.8" has unmet peer dependency "apollo-client@^2.6.3".
warning "workspace-aggregator-58e386bf-df20-49ab-83f3-a64598680545 > @accounts/boost > @graphql-modules/core > @graphql-modules/di@0.7.11" has unmet peer dependency "reflect-metadata@^0.1.12".
warning "workspace-aggregator-58e386bf-df20-49ab-83f3-a64598680545 > @examples/react-graphql-typescript > react-scripts > @typescript-eslint/eslint-plugin@1.13.0" has incorrect peer dependency "eslint@^5.0.0".
warning "workspace-aggregator-58e386bf-df20-49ab-83f3-a64598680545 > @examples/react-graphql-typescript > react-scripts > @typescript-eslint/parser@1.13.0" has incorrect peer dependency "eslint@^5.0.0".
[4/4] Building fresh packages...
Done in 253.00s.
```


## start attempt result


```bash
prishub@ubuntu-one:~/dev/xxx/web/accounts-js/accounts/examples/graphql-server-typescript$ yarn start
yarn run v1.17.3
$ NODE_ENV=development nodemon -w src -x ts-node src/index.ts
[nodemon] 1.19.2
[nodemon] to restart at any time, enter `rs`
[nodemon] watching dir(s): src/**/*
[nodemon] starting `ts-node src/index.ts`

/home/prishub/dev/xxx/web/accounts-js/accounts/node_modules/ts-node/src/index.ts:245
    return new TSError(diagnosticText, diagnosticCodes)
           ^
TSError: ⨯ Unable to compile TypeScript:
src/index.ts:1:33 - error TS2307: Cannot find module '@accounts/database-manager'.

1 import { DatabaseManager } from '@accounts/database-manager';
                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/index.ts:2:32 - error TS2307: Cannot find module '@accounts/graphql-api'.

2 import { AccountsModule } from '@accounts/graphql-api';
                                 ~~~~~~~~~~~~~~~~~~~~~~~
src/index.ts:3:30 - error TS2307: Cannot find module '@accounts/mongo'.

3 import MongoDBInterface from '@accounts/mongo';
                               ~~~~~~~~~~~~~~~~~
src/index.ts:4:34 - error TS2307: Cannot find module '@accounts/password'.

4 import { AccountsPassword } from '@accounts/password';
                                   ~~~~~~~~~~~~~~~~~~~~
src/index.ts:5:32 - error TS2307: Cannot find module '@accounts/server'.

5 import { AccountsServer } from '@accounts/server';
                                 ~~~~~~~~~~~~~~~~~~

    at createTSError (/home/prishub/dev/xxx/web/accounts-js/accounts/node_modules/ts-node/src/index.ts:245:12)
    at reportTSError (/home/prishub/dev/xxx/web/accounts-js/accounts/node_modules/ts-node/src/index.ts:249:19)
    at getOutput (/home/prishub/dev/xxx/web/accounts-js/accounts/node_modules/ts-node/src/index.ts:362:34)
    at Object.compile (/home/prishub/dev/xxx/web/accounts-js/accounts/node_modules/ts-node/src/index.ts:395:32)
    at Module.m._compile (/home/prishub/dev/xxx/web/accounts-js/accounts/node_modules/ts-node/src/index.ts:473:43)
    at Module._extensions..js (internal/modules/cjs/loader.js:947:10)
    at Object.require.extensions.<computed> [as .ts] (/home/prishub/dev/xxx/web/accounts-js/accounts/node_modules/ts-node/src/index.ts:476:12)
    at Module.load (internal/modules/cjs/loader.js:790:32)
    at Function.Module._load (internal/modules/cjs/loader.js:703:12)
    at Function.Module.runMain (internal/modules/cjs/loader.js:999:10)
```

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -21,7 +21,7 @@ The Accounts project was intended - since its inception - to be a community main
 - Install project dependencies: `yarn`
 - Link together all the packages: `yarn setup`
 - Compile the packages `yarn compile`
-- Watch the packages for changes and recompile: `yarn start`
+- Watch the packages for changes and recompile: `yarn start` (You need to run this command in the package subfolder you are updating)
 - If you want to use the accounts project in your own project, use `yarn link @accounts/<name of package>` within your project.
 - Run `docker-compose up -d` to start database services required for tests.
 - Run `yarn test` to run all the tests.

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -20,6 +20,7 @@ The Accounts project was intended - since its inception - to be a community main
 
 - Install project dependencies: `yarn`
 - Link together all the packages: `yarn setup`
+- Compile the packages `yarn compile`
 - Watch the packages for changes and recompile: `yarn start`
 - If you want to use the accounts project in your own project, use `yarn link @accounts/<name of package>` within your project.
 - Run `docker-compose up -d` to start database services required for tests.

--- a/examples/README.md
+++ b/examples/README.md
@@ -1,6 +1,8 @@
-Accounts-js examples
+# Accounts-js examples
 
-This repository provides examples on how to use the [accounts-js](https://github.com/accounts-js/accounts) packages.
+This folder provides examples on how to use the [accounts-js](https://github.com/accounts-js/accounts) packages.
+
+## Examples
 
 REST Server + REST client example:
 
@@ -16,5 +18,3 @@ Other examples:
 
 - Accounts Boost [accounts-boost](./accounts-boost)
 - Typeorm + PostgreSQL example [graphql-server-typeorm-postgres](./graphql-server-typeorm-postgres)
-
-Please go to corresponding packages and see its README.

--- a/examples/graphql-server-typeorm-postgres/README.md
+++ b/examples/graphql-server-typeorm-postgres/README.md
@@ -2,6 +2,16 @@
 
 This example demonstrate how to use [accounts-js](https://github.com/accounts-js/accounts) with PostgreSQL thanks to the amazing [@accounts/typeorm](https://www.npmjs.com/package/@accounts/typeorm) plugin built by (https://github.com/birkir/)[@birkir]for his awesome project: (https://github.com/birkir/prime)[PrimeCMS].
 
+## Setup example
+
+In order to be able to run this example on your machine you first need to do the following steps:
+
+- Clone the repository `git clone git@github.com:accounts-js/accounts.git`
+- Install project dependencies: `yarn`
+- Link together all the packages: `yarn setup`
+- Compile the packages `yarn compile`
+- Go to the example folder `cd examples/graphql-server-typeorm-postgres`
+
 ## Prerequisites
 
 If you're on a mac, you will probably already have postgres installed as a brew service.
@@ -16,12 +26,6 @@ docker-compose up -d
 to start a new one.
 
 ## Getting Started
-
-```bash
-git clone https://github.com/accounts-js/accounts-js.git
-cd examples/graphql-server-typeorm-postgres
-yarn install
-```
 
 Start the app.
 

--- a/examples/graphql-server-typescript/README.md
+++ b/examples/graphql-server-typescript/README.md
@@ -2,6 +2,16 @@
 
 This example demonstrate how to use [accounts-js](https://github.com/accounts-js/accounts).
 
+## Setup example
+
+In order to be able to run this example on your machine you first need to do the following steps:
+
+- Clone the repository `git clone git@github.com:accounts-js/accounts.git`
+- Install project dependencies: `yarn`
+- Link together all the packages: `yarn setup`
+- Compile the packages `yarn compile`
+- Go to the example folder `cd examples/graphql-server-typescript`
+
 ## Prerequisites
 
 You will need a MongoDB server to run this server. If you don't have a MongoDB server running already, and you have Docker & Docker Compose, you can do
@@ -13,12 +23,6 @@ docker-compose up -d
 to start a new one.
 
 ## Getting Started
-
-```bash
-git clone https://github.com/accounts-js/examples.git
-cd examples/graphql-server-typescript
-yarn install
-```
 
 Start the app.
 

--- a/examples/react-graphql-typescript/README.md
+++ b/examples/react-graphql-typescript/README.md
@@ -2,13 +2,17 @@
 
 This example demonstrate how to use [accounts-js](https://github.com/accounts-js/accounts).
 
-## Getting Started
+## Setup example
 
-```bash
-git clone https://github.com/accounts-js/examples.git
-cd examples/react-graphql-typescript
-yarn install
-```
+In order to be able to run this example on your machine you first need to do the following steps:
+
+- Clone the repository `git clone git@github.com:accounts-js/accounts.git`
+- Install project dependencies: `yarn`
+- Link together all the packages: `yarn setup`
+- Compile the packages `yarn compile`
+- Go to the example folder `cd examples/react-graphql-typescript`
+
+## Getting Started
 
 If you haven't started the example server side, [go ahead to start it first](../graphql-server-typescript).
 

--- a/examples/react-rest-typescript/README.md
+++ b/examples/react-rest-typescript/README.md
@@ -4,13 +4,17 @@ This example demonstrate how to use [accounts-js](https://github.com/accounts-js
 
 In order to get the example running you also need to start the [REST server](https://github.com/accounts-js/accounts/tree/master/examples/rest-express-typescript).
 
-## Getting Started
+## Setup example
 
-```
-git clone https://github.com/accounts-js/accounts.git
-cd examples/react-rest-typescript
-yarn install
-```
+In order to be able to run this example on your machine you first need to do the following steps:
+
+- Clone the repository `git clone git@github.com:accounts-js/accounts.git`
+- Install project dependencies: `yarn`
+- Link together all the packages: `yarn setup`
+- Compile the packages `yarn compile`
+- Go to the example folder `cd examples/react-rest-typescript`
+
+## Getting Started
 
 Start the app.
 

--- a/examples/rest-express-typescript/README.md
+++ b/examples/rest-express-typescript/README.md
@@ -4,13 +4,17 @@ This example demonstrate how to use [accounts-js](https://github.com/accounts-js
 
 You must have a mongodb server running before starting the server.
 
-## Getting Started
+## Setup example
 
-```
-git clone https://github.com/accounts-js/examples.git
-cd examples/rest-express-typescript
-yarn install
-```
+In order to be able to run this example on your machine you first need to do the following steps:
+
+- Clone the repository `git clone git@github.com:accounts-js/accounts.git`
+- Install project dependencies: `yarn`
+- Link together all the packages: `yarn setup`
+- Compile the packages `yarn compile`
+- Go to the example folder `cd examples/rest-express-typescript`
+
+## Getting Started
 
 Start the app.
 

